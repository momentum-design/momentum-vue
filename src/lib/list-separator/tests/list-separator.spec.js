import { mount } from '@vue/test-utils';
import ListSeparator from '../index.vue';

describe('ListSeparator', () => {
  it('should match snapshot', () => {
    const wrapper = mount(ListSeparator);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render one ListSeparator', () => {
    const wrapper = mount(ListSeparator);
    expect(wrapper.classes('md-list-separator')).toBeTruthy();
  });

  it('should render text prop', () => {
    const wrapper = mount(ListSeparator, {
      propsData: {
        text: 'Today'
      }
    });

    expect(wrapper.findAll('.md-list-separator__text').length).toEqual(1);
    expect(wrapper.find('.md-list-separator__text').text()).toEqual('Today');
  });

  it('should change the line color', () => {
    const wrapper = mount(ListSeparator, {
      propsData: {
        lineColor: 'red'
      }
    });

    expect(wrapper.attributes().style).toEqual('color: red;');
  });

  it('should add padding to the text prop', () => {
    const wrapper = mount(ListSeparator, {
      propsData: {
        text: 'Today',
        textPadding: '40px'
      }
    });

    expect(wrapper.find('.md-list-separator__text').attributes().style).toEqual('padding: 40px;');
  });

  it('should change the text color', () => {
    const wrapper = mount(ListSeparator, {
      propsData: {
        text: 'Today',
        textColor: 'green'
      }
    });

    expect(wrapper.find('.md-list-separator__text').attributes().style).toEqual('color: green;');
  });

  it('should handle class prop', () => {
    const wrapper = mount(ListSeparator, {
      propsData: {
        class: 'right'
      }
    });

    expect(wrapper.findAll('.right').length).toEqual(1);
  });

  it('should render children', () => {
    const wrapper = mount(ListSeparator, {
      slots: {
        default: '<div>Test</div>'
      }
    });
    expect(wrapper.element.__vue__.$children.length).toEqual(1);
  });
});

