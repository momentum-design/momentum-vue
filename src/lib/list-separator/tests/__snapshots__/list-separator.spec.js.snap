// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSeparator should match snapshot 1`] = `<div class="md-list-separator"><span class="md-list-separator__container"><!----></span></div>`;
